domain/consensus/processes/consensusstatemanager/resolve_block_status_test.go

```diff
@@ -156,124 +158,139 @@ func TestDoubleSpends(t *testing.T) {
+			t.Fatalf("Error getting blockC: %+v", err)
 		}
 		fees := uint64(1)
-		transactionToSpend0 := fundingBlock3.Transactions[transactionhelper.CoinbaseTransactionIndex]
-		spendingTransaction0, err := testutils.CreateTransaction(transactionToSpend0, fees)
+		transactionFromBlockC := blockC.Transactions[transactionhelper.CoinbaseTransactionIndex]
+		// transactionFromRedBlock is spending TransactionFromBlockC.
+		transactionFromRedBlock, err := testutils.CreateTransaction(transactionFromBlockC, fees)
 		if err != nil {
-			t.Fatalf("Error creating a transaction: %+v", err)
+			t.Fatalf("Error creating a transactionFromRedBlock: %+v", err)
 		}
-		spendingTransaction0Input0UTXOEntry, err := testConsensus.ConsensusStateStore().
-			UTXOByOutpoint(testConsensus.DatabaseContext(), stagingArea, &spendingTransaction0.Inputs[0].PreviousOutpoint)
+		transactionFromRedBlockInput0UTXOEntry, err := testConsensus.ConsensusStateStore().
+			UTXOByOutpoint(testConsensus.DatabaseContext(), stagingArea, &transactionFromRedBlock.Inputs[0].PreviousOutpoint)
 		if err != nil {
-			t.Fatalf("Error getting UTXOEntry for spendingTransaction0Input0: %s", err)
+			t.Fatalf("Error getting UTXOEntry for transactionFromRedBlockInput: %s", err)
 		}
-		redHash, _, err := testConsensus.AddBlock([]*externalapi.DomainHash{fundingBlock3Hash}, nil,
-			[]*externalapi.DomainTransaction{spendingTransaction0})
+		redHash, _, err := testConsensus.AddBlock([]*externalapi.DomainHash{blockHashC}, nil,
+			[]*externalapi.DomainTransaction{transactionFromRedBlock})
 		if err != nil {
 			t.Fatalf("Error creating redBlock: %+v", err)
 		}
 
-		transactionToSpend1 := spendingTransaction0
-		spendingTransaction1, err := testutils.CreateTransaction(transactionToSpend1, fees)
+		transactionFromBlueBlock, err := testutils.CreateTransaction(transactionFromRedBlock, fees)
 		if err != nil {
-			t.Fatalf("Error creating spendingTransaction1: %+v", err)
+			t.Fatalf("Error creating transactionFromBlueBlock: %+v", err)
 		}
-		spendingTransaction1Input0UTXOEntry, err := testConsensus.ConsensusStateStore().
-			UTXOByOutpoint(testConsensus.DatabaseContext(), stagingArea, &spendingTransaction1.Inputs[0].PreviousOutpoint)
+		transactionFromBlueBlockInput0UTXOEntry, err := testConsensus.ConsensusStateStore().
+			UTXOByOutpoint(testConsensus.DatabaseContext(), stagingArea, &transactionFromBlueBlock.Inputs[0].PreviousOutpoint)
 		if err != nil {
-			t.Fatalf("Error getting UTXOEntry for spendingTransaction1Input0: %s", err)
+			t.Fatalf("Error getting UTXOEntry for transactionFromBlueBlockInput: %s", err)
 		}
-		blueHash, _, err := testConsensus.AddBlock(
-			[]*externalapi.DomainHash{tipHash, redHash},
-			nil,
-			[]*externalapi.DomainTransaction{spendingTransaction1})
+		blueBlockScriptPublicKey := &externalapi.ScriptPublicKey{Script: []byte{3}, Version: 0}
+		blueHash, _, err := testConsensus.AddBlock([]*externalapi.DomainHash{lastBlockInChain, redHash},
```

Comment:
Maybe you should call it `blueChildOfRedBlock` to differentiate it from all of the other blue blocks?